### PR TITLE
FO: Fix price display in case of quantity specific price

### PR DESCRIPTION
**Pull request by erouvier29**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Fix price display when quantity discount on combination
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-7797
| How to test?  | Cf. below

Additional commit to #7644:
Fixes quantity discount on all combinations case
Fixes alternate currency case

How to test:
- define a second currency
- define a fixed specific price for all combinations of a product from quantity 2
- goto product page and select the alternate currency
- volume discount table is incorrect
- set quantity to 2, product price is incorrect

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/7869)

### CODE REVIEW

**Comment by kmorgen** [comment]
This fix the quantity discount listing on the product pages again, thanks!
However will still be messed up if changing currency. Also if using discount amount/percentage to make and quantity discount for a specific group, it will display correct values (for default currency), but calculate wrong in cart. Seem to deduct the discount from the retail price, not the 1pcs price for that particular group.

Please refer to: http://forge.prestashop.com/browse/PSCSX-9037

**Comment by marionf** [comment]
Hello @erouvier29 
Can you please complete the "how to test" section with the steps ?
The scenario described in the ticket PSCSX-7797 is already solved by PR #7644
Thanks

**Comment by erouvier29** [comment]
@marionf: done
Regards

**Comment by marionf** [comment]
Hello @erouvier29 

I think the price isn't good when there is a fixed specific price.
1) I import the united states pack localization (exchange rate 1.1639)
2) I create a product with a price of 100€ tax included.
3) I create a fixed specific price of 50€ tax included discount from quantity 2
4) The price displayed on the product page is 58.19 USD tax included instead of 58.20
pr7869

pr7869-3


In the cart the price displayed is well 58.20 USD tax included
pr7869-5



The same scenario with a specific price of 50% discount from quantity 2 is good, the price displayed on the product page is well 58.20 USD tax included
pr7869-4

pr7869-2

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -334,9 +334,8 @@ protected function assignPriceAndTax()
 
         $quantity_discounts = SpecificPrice::getQuantityDiscounts($id_product, $id_shop, $id_currency, $id_country, $id_group, null, true, (int)$this->context->customer->id);
         foreach ($quantity_discounts as &$quantity_discount) {
-            if ($quantity_discount['id_product_attribute']) {
-                $quantity_discount['base_price'] = $this->product->getPrice(Product::$_taxCalculationMethod == PS_TAX_INC, $quantity_discount['id_product_attribute']);
-
+            $quantity_discount['base_price'] = $this->product->getPrice(Product::$_taxCalculationMethod == PS_TAX_INC, $quantity_discount['id_product_attribute']);
+        	if ($quantity_discount['id_product_attribute']) {
```
Comment (maximebiloe) [inline controllers/front/ProductController.php]: There is an extra tab here. Can you remove it, please?

File: themes/default-bootstrap/js/product.js [hunk 2]
```diff
@@ -944,9 +944,9 @@ function updateDiscountTable(newPrice)
 		}
 
 		if (displayDiscountPrice != 0)
-			$(this).children('td').eq(1).text( formatCurrency(discountedPrice * currencyRate, currencyFormat, currencySign, currencyBlank) );
-		$(this).children('td').eq(2).text(upToTxt + ' ' + formatCurrency(discountUpTo * currencyRate, currencyFormat, currencySign, currencyBlank));
-	});
+			$(this).children('td').eq(1).text( formatCurrency(discountedPrice, currencyFormat, currencySign, currencyBlank) );
+		$(this).children('td').eq(2).text(upToTxt + ' ' + formatCurrency(discountUpTo, currencyFormat, currencySign, currencyBlank));
+});
```
Comment (maximebiloe) [inline themes/default-bootstrap/js/product.js]: Missing a tab here

**Review by maximebiloe — CHANGES_REQUESTED** [review]

**Comment by erouvier29** [comment]
Indentation corrected.
Rounding issue reported by @marionf still present.

**Comment by hibatallahAouadni** [comment]
@erouvier29 could you please rebase it and resolve the conflits. 
Thanks in advance.

**Pull request closed without merging**